main.go

```diff
@@ -141,6 +150,20 @@ func main() {
 
 	if err := etcdCopyBackupsTask.SetupWithManager(mgr, etcdCopyBackupsTaskWorkers); err != nil {
 		setupLog.Error(err, "Unable to create controller", "Controller", "EtcdCopyBackupsTask")
+	}
+
+	lc, err := controllers.NewLeaseControllerWithImageVector(mgr, controllersconfig.CompactionConfig{
```

Comment:
* We have "lease controller" but "compaction config". I would propose to harmonize the naming, that is use either "lease" or "compaction" (or perhaps `CompactionLease`), but not both.
* Shouldn't we be able to configure the number of workers for this controller, similarly to other controllers?

Reply:
I didn't understand the second point. `if err := lc.SetupWithManager(mgr, compactionWorkers); err != nil {` is there at line no ~160. The `compactionWorkers` is configurable. Am I missing something ?

Reply:
Hmm.. For the first point, we introduced lease controller because it will handle all the leases (like member leases in future for multi node ETCD). So we should name our controller as lease controller. But the configs are currently only for compaction job. But in future, it may hold config for other things that are dependent on lease. We can change the naming of this config. Probably, `leaseConfig` ?

Reply:
> For the first point, we introduced lease controller because it will handle all the leases (like member leases in future for multi node ETCD)

I'm a little bit surprised, TBH. Why would we do we want to do that?

Reply:
> I didn't understand the second point. if err := lc.SetupWithManager(mgr, compactionWorkers); err != nil { is there at line no ~160. The compactionWorkers is configurable. Am I missing something ?

You are right, everything is ok actually. I was again misled by the naming mismatch between "lease" and "compaction", so it wasn't immediately obvious that `compactionWorkers` is actually used with the "lease controller". Hopefully this could be improved.

Reply:
Done